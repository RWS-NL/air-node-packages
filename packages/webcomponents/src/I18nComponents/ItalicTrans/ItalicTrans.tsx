import React, { FC, memo, Fragment } from 'react';
import { Trans } from 'react-i18next';

export interface ItalicTransProps {
  titleKey: string;
  italicKey: string;
}

export const ItalicTrans: FC<ItalicTransProps> = ({ titleKey, italicKey }) => (
  <Fragment>
    <Trans i18nKey={titleKey} />
    <em><Trans i18nKey={italicKey} /></em>
  </Fragment>
);

export default memo(ItalicTrans);