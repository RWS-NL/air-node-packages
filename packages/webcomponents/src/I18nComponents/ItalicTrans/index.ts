export { default as ItalicTrans, ItalicTransProps } from './ItalicTrans';