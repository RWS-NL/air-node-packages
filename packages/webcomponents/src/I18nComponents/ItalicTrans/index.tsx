import React, { Fragment, memo } from 'react';
import { Trans } from 'react-i18next';

export interface ItalicTransProps {
  titleKey: string;
  italicKey: string;
}

/**
 * Creates a Trans body with one part regular and one part *italic*
 * @param param Props to pass to the ItalicTrans component
 * @example
 * ```jsx
 * <ItalicTrans titleKey='sample.key.normal' italicKey='sample.key.italic'/>
 * ```
 */
export const ItalicTrans = memo(({ titleKey, italicKey }: ItalicTransProps) => (
  <Fragment>
    <Trans i18nKey={titleKey} />
    <em>
      <Trans i18nKey={italicKey} />
    </em>
  </Fragment>
));

export default ItalicTrans;
