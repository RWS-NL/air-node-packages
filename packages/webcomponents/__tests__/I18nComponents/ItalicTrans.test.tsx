import { shallow, ShallowWrapper } from 'enzyme';
import React from 'react';
import { Trans } from 'react-i18next';
import ItalicTrans, { ItalicTransProps } from '../../src/I18nComponents/ItalicTrans';

describe('Component Testing', () => {
  let italicTrans: ShallowWrapper<ItalicTransProps>;

  beforeAll(() => {
    italicTrans = shallow(<ItalicTrans titleKey='test' italicKey='test' />);
  });

  test('GIVEN base configuration THEN should have two <Trans> components', () => {
    expect(italicTrans.find('em')).toHaveLength(1);
    expect(italicTrans.find(Trans)).toHaveLength(2);
  });
});

describe('Snapshot Testing', () => {
  test('Required Props', () => {
    const italicTrans = shallow(<ItalicTrans titleKey='test' italicKey='test' />);
    expect(italicTrans).toMatchSnapshot();
  });
});
