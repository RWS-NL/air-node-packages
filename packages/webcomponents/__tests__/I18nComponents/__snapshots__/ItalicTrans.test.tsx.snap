// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Testing Required Props 1`] = `
<Fragment>
  <Trans
    i18nKey="test"
  />
  <em>
    <Trans
      i18nKey="test"
    />
  </em>
</Fragment>
`;
